samples/components/oauth/redshift.azure.xml

```diff
@@ -64,7 +68,7 @@
     </entry>
     <entry>
       <key>username</key>
-      <value>preferred_username</value>
```

Comment:
why we decided to change the mapping here? if we keep scope "profile", we should have preferred_username info

Reply:
we have been using this mapping for other azure connectors: https://gitlab.tableausoftware.com/connectivity/plugins/azure-sqldb/-/blob/master/src/script/oauthConfig.xml?ref_type=heads#L75

Reply:
For Okta it may be okay to use either. For Azure, in particular for IAM IDC they have to use v1 Azure tokens which don't have preferred_username claim. [https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference](https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference)

> The primary username that represents the user. It could be an email address, phone number, or a generic username without a specified format. Its value is mutable and might change over time. Since it's mutable, this value can't be used to make authorization decisions. It can be used for username hints and in human-readable UI as a username. The profile scope is required to receive this claim. Present only in v2.0 tokens.

Reply:
I can revert it from this one just to show you can use either.